src/xmipp/libraries/reconstruction/pdb_label_from_volume.cpp

```diff
@@ -0,0 +1,328 @@
+		STARTINGX(inputVol) = 0;
+
+		STARTINGZ(inputMask) = 0;
+		STARTINGY(inputMask) = 0;
+		STARTINGX(inputMask) = 0;
+	}
+
+}
+
+/* Compute protein geometry ------------------------------------------------ */
+void ProgPdbValueToVol::computeProteinGeometry()
+{
+    std::ifstream fh_pdb;
+    std::ofstream fh_out(fn_out);
+    fh_pdb.open(fn_pdb.c_str());
+
+	MetaData mdmean;
+	size_t objId;
+	objId = mdmean.addObject();
+
+    if (!fh_pdb)
+        REPORT_ERROR(ERR_IO_NOTEXIST, fn_pdb);
+
+    double suma=0, sumaP=0;
+    int numA=0;
+
+
+    while (!fh_pdb.eof())
+    {
+        // Read an ATOM line
+        std::string line;
+        getline(fh_pdb, line);
+        if (line == "")
+        {
+    		fh_out << line << " \n";
+            continue;
+        }
+        std::string kind = line.substr(0,4);
+        if (kind != "ATOM" && kind !="HETA")
+        {
+    		fh_out << line << " \n";
+            continue;
+        }
+
+        // Extract atom type and position
+        // Typical line:
+        // ATOM    909  CA  ALA A 161      58.775  31.984 111.803  1.00 34.78
+        std::string atom_type = line.substr(13,2);
+        double x = textToFloat(line.substr(30,8));
+        double y = textToFloat(line.substr(38,8));
+        double z = textToFloat(line.substr(46,8));
+
+        // Correct position
+        Matrix1D<double> r(3);
+        VECTOR_R3(r, x, y, z);
+        r *= 1/Ts;
+
+        // Characterize atom
+        double radius2=radius*radius;
+
+        // Find the part of the volume that must be updated
+        int k0 = XMIPP_MAX(FLOOR(ZZ(r) - radius), STARTINGZ(inputVol));
+        int kF = XMIPP_MIN(CEIL(ZZ(r) + radius), FINISHINGZ(inputVol));
+        int i0 = XMIPP_MAX(FLOOR(YY(r) - radius), STARTINGY(inputVol));
+        int iF = XMIPP_MIN(CEIL(YY(r) + radius), FINISHINGY(inputVol));
+        int j0 = XMIPP_MAX(FLOOR(XX(r) - radius), STARTINGX(inputVol));
+        int jF = XMIPP_MIN(CEIL(XX(r) + radius), FINISHINGX(inputVol));
+
+        int ka = XMIPP_MAX(FLOOR(ZZ(r)), STARTINGZ(inputVol));
+        int ia = XMIPP_MAX(FLOOR(YY(r)), STARTINGZ(inputVol));
+        int ja = XMIPP_MAX(FLOOR(XX(r)), STARTINGZ(inputVol));
+
+
+        float atomS=0.0;
+        float atomP=0.0;
+        float atomN=0.0;
+        float value=0.0;
+        int cont=0;
+        for (int k = k0; k <= kF; k++)
+        {
+            double zdiff=ZZ(r) - k;
+            double zdiff2=zdiff*zdiff;
+            for (int i = i0; i <= iF; i++)
+            {
+                double ydiff=YY(r) - i;
+                double zydiff2=zdiff2+ydiff*ydiff;
+                for (int j = j0; j <= jF; j++)
+                {
+                    double xdiff=XX(r) - j;
+                    double rdiffModule2=zydiff2+xdiff*xdiff;
+                	if (withMask)
+                	{
+						if ( (rdiffModule2<radius2 || (k==ka && i==ia && j==ja)) && (inputMask(k, i , j)>0.00001) )
+						{
+
+							atomS += A3D_ELEM(inputVol,k, i, j);
+							++cont;
+
```

Comment:
inappropriate indention. Check indentions also in the rest of the code.

Reply:
@dmaluenda, in eclipse this works out for me. I dont know?

Reply:
@dmaluenda puedes chequear esto. A mi me sale bien en eclipse

Reply:
https://stackoverflow.com/a/408449/5484355

Reply:
Thank you @DStrelak. What indentation size? 2 or 4?

Reply:
We don't have any strong rules AFAIK, but I would go for 4.